### ISSUE TITLE
Alexa doesn't discover

### ISSUE THREAD

**Issue by Pseudomizer** [bwssytems/ha-bridge#278]
Hello guys,
I updated to HA 3.5.1 today. I also added 2 new devices in Vera and the HA bridge. Alexa, couldn't find them. I deleted all devices on Alexa. Tried to rediscover. Nothing.
Rebootet both Alexas in my home. Rebootet the HA bridge on my PI. Nothing.
Downgraded to 3.2.1 again. Still not working. 
Disabled uPNP on my router and hammered the IP of Alexa as the uPNP address into the HA bridge just to test. Nothing.
Alexa finds the real Hue bridge with my 12 bulbs and my Nest and my Ecobee3 via the 2 skills. Nothing else is being discovered.
Then I read about the new support for Google home and I changed the port from 5050 to 80 for the HA bridge. 
GOOGLE HOME FOUND ALL DEVICES! Alexa discovers nothing!
Enabled logging on router and HA bridge as suggested in the previous issues thread with one debug level while all others are on info only. My router doesn't show any uPNP logs (Asus RT-A5300) and the HA bridge doesn't log anything either about Alexa... BUT it shows how Google Home requests an updated list every min or two and the HA bridge responds.
At this stage I am out of ideas guys. 

Thanks in advance for any help here.

**Comment by vageesh79** [comment]
same with me.. Every thing was was working 5 days back.. Then it stopped working. Updated to 3.5.1 not alexa not discovering anything. Tried everything. Need some help.

**Comment by Pseudomizer** [comment]
I even removed all devices from the bridge after a backup. Then I added 1 switch and 1 scene just in case the number of devices are causing the issue, which was the case before. Still nothing.

**Comment by vageesh79** [comment]
I tested again today removed everything and tried again. Ha-bridge is discovered by harmony hub but not by alexa app. Anybody has any ideas.

**Comment by jlekhter** [comment]
I'm assuming you already verified that the UPNP address hasn't changed for the device you are using to run your hub and that your router is redirecting to that IP from Amazon.

**Comment by Pseudomizer** [comment]
What are the steps to verify that? I did upgrade the HA bridge to 3.5.1, but I also upgraded Rasperian to the latest version. If that causes changes, then please let me know, how to fix that. 

Thanks in advance.

**Comment by jlekhter** [comment]
What's the IP of your Raspberry Pi? Is it using dhcp or static? If dhcp is it reserved on your router to be the same IP all the time? If you find the IP of your Pi, then set the upnp IP address in the bridge setup to the same address. 

If that doesn't work, I also had to open port 1900 for UDP traffic from my router and point it to the server running the bridge.

**Comment by Pseudomizer** [comment]
I am using DHCP but I reserved the mac address to have the same IP every single time. So the IP is basically fixed to ensure consistency. 

I understand your point about port 1900 but why is Google Home finding the HA bridge and Amazon Echo does not? Is this an Amazon specific port?

**Comment by jlekhter** [comment]
I haven't played around with Google home so I don't know. 

So is your UPNP address the same as your server address? 

If that doesn't resolve try opening up 1900. Otherwise I'm at a loss as well. 

Sorry.

**Comment by Pseudomizer** [comment]
I just created a new rule for 192.168.1.0 where all UDP Traffic gets routed to 192.168.1.50, which is my PI. Still no luck.

**Comment by jlekhter** [comment]
And the UPNP address is 192.168.1.50 in your config?

The rule I have is all inbound traffic on 1900 UDP is sent to my PI. I'm not home so can't check the exact terminology on the router.

**Comment by Pseudomizer** [comment]
Yes

**Comment by vageesh79** [comment]
Pseudomizer - sorry for keeo intrupting in your thread but i am also in same issue. My harmony and anymote app on android both detects ha-bridge running on pi and can control it also but amazon echo cannot find anything. Tried on 2 amazon echo's same. All was working on the old jar file as i was using a very old ha-bridge update and everything was working fine till 4 days back, Then i updated to new jar file and change the config as given in the new updated page now echo not finding anything. Can't figure out anything.

**Comment by Pseudomizer** [comment]
Did you try to downgrade and see, if it works for you? Doesn't work for me, but maybe it works for you.

**Comment by vageesh79** [comment]
Have done that also same here.. Its working great from harmony hub and my phone.

**Comment by simondsmason** [comment]
Add me to the list.  I upgrade to 3.5.1 and then tried to discover some new devices - nothing.  Tried deleting all devices and starting from scratch - nothing discovers.  Everything appears to be running and no errors.  I don't think this is related to 3.5.1 - I think it just started happening at the same time.    My ha-bridge is running on a Windows 7 machine and was working fine up until I started having this issue.  Anything else I can post to help diagnose this?  Thanks.

**Comment by Pseudomizer** [comment]
Here are some logs after going into Debug mode for most of the modules. This clearly shows how the HA bridge is exchanging information with my Google Home device. I pressed discover 3 times during that time and the log doesn't show anything from any of my two Alexa IP addresses. It seems like Alexa doesn't even try to communicate with the HA bridge at all...

11-27-2016 23:44:55.283	DEBUG	isSSDPDiscovery Got SSDP packet from 192.168.1.238:32820, body: M-SEARCH * HTTP/1.1
 HOST: 239.255.255.250:1900
 MAN: "ssdp:discover"
 MX: 2
 ST: ssdp:all
 
	com.bwssystems.HABridge.upnp.UpnpListener
11-27-2016 23:44:55.283	DEBUG	isSSDPDiscovery found message to be valid under strict rules - strict: true	com.bwssystems.HABridge.upnp.UpnpListener
11-27-2016 23:44:55.287	DEBUG	sendUpnpResponse discovery responseTemplate1 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.1.50:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: B827EBFFFE053F93
 ST: upnp:rootdevice
 USN: uuid:2f402f80-da50-11e1-9b23-b827eb053f93::upnp:rootdevice
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-27-2016 23:44:55.287	DEBUG	Sending response string: <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.1.50:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: B827EBFFFE053F93
 ST: upnp:rootdevice
 USN: uuid:2f402f80-da50-11e1-9b23-b827eb053f93::upnp:rootdevice
 
 >>>	com.bwssystems.util.UDPDatagramSender
11-27-2016 23:44:55.288	DEBUG	sendUpnpResponse discovery responseTemplate2 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.1.50:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: B827EBFFFE053F93
 ST: uuid:2f402f80-da50-11e1-9b23-b827eb053f93
 USN: uuid:2f402f80-da50-11e1-9b23-b827eb053f93
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-27-2016 23:44:55.289	DEBUG	Sending response string: <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.1.50:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: B827EBFFFE053F93
 ST: uuid:2f402f80-da50-11e1-9b23-b827eb053f93
 USN: uuid:2f402f80-da50-11e1-9b23-b827eb053f93
 
 >>>	com.bwssystems.util.UDPDatagramSender
11-27-2016 23:44:55.290	DEBUG	sendUpnpResponse discovery responseTemplate3 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.1.50:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: B827EBFFFE053F93
 ST: urn:schemas-upnp-org:device:basic:1
 USN: uuid:2f402f80-da50-11e1-9b23-b827eb053f93
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-27-2016 23:44:55.290	DEBUG	Sending response string: <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.1.50:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: B827EBFFFE053F93
 ST: urn:schemas-upnp-org:device:basic:1
 USN: uuid:2f402f80-da50-11e1-9b23-b827eb053f93
 
 >>>	com.bwssystems.util.UDPDatagramSender

**Comment by bwssytems** [comment]
Dumb question. Did anyone restart their Echo? And what type of Echo's are they?

**Comment by Pseudomizer** [comment]
Lol. Yes, and I even went as far rebooting both Alexas at the same time and shutting one off and leave just one on. They are both full blown Echos. No tap or dots.

**Comment by bwssytems** [comment]
Ok, then. your log shows that it did not send a upnp multicast request out or it would have been seen by the bridge.

**Comment by Pseudomizer** [comment]
When I remove the real hue bridge from Alexa and then go to discover, it gets found. So I assume that Alexa is sending a broadcast out. 
It is not showing up in the logs of the HA bridge though. 
So I don't know what else to check.

**Comment by bwssytems** [comment]
Ahh, yes. When there is a real hue bridge on the network, you need to disconnect that and then discover for the ha-bridge, then reconnect the real hue bridge. Known issue with the Alexa.

**Comment by vageesh79** [comment]
Tried restarting both amazon echo. Don't have any real hue bridge on network. Ha-bridge is discovered by harmony hub and Anymote just not by echo. Is it possible that something in echo got updated as before that everything was working.

**Comment by bwssytems** [comment]
@vageesh79 Have you turned on trace upnp in the settings to see what is talking to your bridge? That will help immensely.

**Comment by vageesh79** [comment]
Here is the Upnp trace log i can see -

11-28-2016 12:45:57.791	INFO	== Spark has ignited ...	spark.webserver.JettySparkServer
11-28-2016 12:45:57.808	INFO	>> Listening on 0.0.0.0:80	spark.webserver.JettySparkServer
11-28-2016 12:45:58.217	INFO	HABridge device management service started....	com.bwssystems.HABridge.devicemanagmeent.DeviceResource
11-28-2016 12:45:58.269	INFO	Hue description service started....	com.bwssystems.HABridge.upnp.UpnpSettingsResource
11-28-2016 12:45:58.280	INFO	Initializing UDP response Seocket...	com.bwssystems.util.UDPDatagramSender
11-28-2016 12:45:58.294	INFO	UDP response Seocket initialized to: 50000	com.bwssystems.util.UDPDatagramSender
11-28-2016 12:45:58.309	INFO	Traceupnp: upnp device settings requested: from 192.168.0.20:80	com.bwssystems.HABridge.upnp.UpnpSettingsResource
11-28-2016 12:45:58.321	INFO	Traceupnp: upnp device settings template filled with address: 192.168.0.210 and port: 80	com.bwssystems.HABridge.upnp.UpnpSettingsResource
11-28-2016 12:45:58.511	INFO	Hue emulator service started....	com.bwssystems.HABridge.hue.HueMulator
11-28-2016 12:45:58.532	INFO	UPNP Discovery Listener starting....	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:45:58.545	INFO	Traceupnp: eth0 ... has addr /fe80:0:0:0:4522:60ca:bdd7:1bb9%eth0	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:45:58.551	INFO	Traceupnp: wlan0 ... has addr /fe80:0:0:0:6c37:7945:407:d636%wlan0	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:45:58.558	INFO	Traceupnp: wlan0 ... has addr /fd00:f0f2:49a7:5d22:57d7:576a:c214:3280%wlan0	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:45:58.567	INFO	Traceupnp: wlan0 ... has addr /192.168.0.210	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:45:58.577	INFO	Traceupnp: Adding lo to our interface set	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:34.129	INFO	Traceupnp: isSSDPDiscovery found message to be valid under strict rules - strict: true	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:34.266	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate1 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: upnp:rootdevice
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086::upnp:rootdevice
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:34.305	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate2 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:34.334	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate3 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: urn:schemas-upnp-org:device:basic:1
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:39.150	INFO	Traceupnp: isSSDPDiscovery found message to be an M-SEARCH message.	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:39.160	INFO	Traceupnp: isSSDPDiscovery found message to be valid under strict rules - strict: true	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:39.161	INFO	Traceupnp: SSDP packet from 192.168.0.210:43699, body: M-SEARCH * HTTP/1.1
 HOST: 239.255.255.250:1900
 MAN: "ssdp:discover"
 MX: 1
 ST: upnp:rootdevice
 
	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:39.197	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate1 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: upnp:rootdevice
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086::upnp:rootdevice
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:39.216	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate2 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 12:46:39.233	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate3 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: urn:schemas-upnp-org:device:basic:1
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener

**Comment by bwssytems** [comment]
Now, what is the address of your echo?

**Comment by vageesh79** [comment]
Its 192.168.0.22. It is connected to internet and my other skills are working on that but i think echo is not finding the bridge

**Comment by vageesh79** [comment]
192.168.0 210 is the raspberry pi address running ha-bridge..

**Comment by bwssytems** [comment]
Did you try telling it to do a discover multiple times after the bridge restart?

**Comment by vageesh79** [comment]
Just tried again and this is what i gt in logs - 

11-28-2016 13:47:16.878	INFO	Traceupnp: isSSDPDiscovery found message to be an M-SEARCH message.	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:16.885	INFO	Traceupnp: isSSDPDiscovery found message to be valid under strict rules - strict: true	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:16.885	INFO	Traceupnp: SSDP packet from 192.168.0.210:45271, body: M-SEARCH * HTTP/1.1
 HOST: 239.255.255.250:1900
 MAN: "ssdp:discover"
 MX: 1
 ST: ssdp:all
 
	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:16.912	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate1 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: upnp:rootdevice
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086::upnp:rootdevice
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:16.913	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate2 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:16.997	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate3 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: urn:schemas-upnp-org:device:basic:1
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:21.880	INFO	Traceupnp: isSSDPDiscovery found message to be an M-SEARCH message.	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:21.887	INFO	Traceupnp: isSSDPDiscovery found message to be valid under strict rules - strict: true	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:21.888	INFO	Traceupnp: SSDP packet from 192.168.0.210:57928, body: M-SEARCH * HTTP/1.1
 HOST: 239.255.255.250:1900
 MAN: "ssdp:discover"
 MX: 1
 ST: upnp:rootdevice
 
	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:21.892	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate1 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: upnp:rootdevice
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086::upnp:rootdevice
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:21.911	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate2 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 
 >>>	com.bwssystems.HABridge.upnp.UpnpListener
11-28-2016 13:47:21.914	INFO	Traceupnp: sendUpnpResponse discovery responseTemplate3 is <<<HTTP/1.1 200 OK
 HOST: 239.255.255.250:1900
 CACHE-CONTROL: max-age=100
 EXT:
 LOCATION: http://192.168.0.210:80/description.xml
 SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0
 hue-bridgeid: 30B5C2FFFE186086
 ST: urn:schemas-upnp-org:device:basic:1
 USN: uuid:2f402f80-da50-11e1-9b23-30b5c2186086
 
 >>>	com.b

**Comment by bwssytems** [comment]
Well, it sseems the echo is not broadcasting a upnp multicast message. Check your router settings.

**Comment by vageesh79** [comment]
Let me check that again.. upnp is open on router. I even opened the port 1900 also but il see if i missed anything.

**Comment by jlekhter** [comment]
I noticed that both lan0 and wan0 have IPv6 addresses bound as well. Have you tried setting Java to prefer ipv4v?

**Comment by jlekhter** [comment]
-Djava.net.preferIPv4Stack=true

**Comment by bwssytems** [comment]
It is automatically set to IPV4 in the code.

**Comment by vageesh79** [comment]
Cannot find anything on router. I just tried enabling vera and harmony skills and echo is discovering vera and harmony devices. I think i had to start from scratch installing everything again.

**Comment by bigbasec** [comment]
Same problem here.  Decided to give this a go to replace an original Amazon-ha-bridge install.  I should have stuck with "if it ain't broke don't fix it", as now I can't discover any devices or anything.  Nothing on the router changed at all, just rebooted it, as well as everything else and no change.

**Comment by Pseudomizer** [comment]
My echos have 192.168.1.34 and .35. My Asus router has uPnp enabled. 

I also noticed one more phenomenon. Vera uses uPnp to discover Sonos. This used to work but now the discovery returns nothing. I have to manually enter the IP addresses of my 3 Sonos speakers. 

This is getting frustrating. uPnP, which should be universal plug and play becomes plug and pray now. I can troubleshoot IP issues but I have never troubleshooted uPnP issues. 

In summary:
Both Alexas don't find the HA bridge. 
Google home DOES find the HA bridge. 
Sonos is no longer discoverable but works with manual IP entering.

**Comment by vageesh79** [comment]
Anyone else facing issues

**Comment by Pseudomizer** [comment]
I am counting 4 people above having the same issues in total.

**Comment by bigbasec** [comment]
I actually resolved mine, turned out that on restart my settings didn't
save, so I needed to go into the settings page and set uPNP IP again, saved
it restarted and did a rediscover.  All was well after that.

**Comment by vageesh79** [comment]
Today i tried it on fresh installation used an old version but still same. I think the problem is with Echo only as if i add hue bridge on harmony hub it is added. Only Echo is having trouble finding the bridge. Anyone has any idea. UPNP is working fine on network as harmony and my phone software Anymote are finding bridge easily.

**Comment by bwssytems** [comment]
You might want to look into wireshark to debug your network.

**Comment by vageesh79** [comment]
Got it working. Problem was with the router only. I switched to a different gateway and now echo is discovering.

**Comment by bwssytems** [comment]
Great!

**Comment by Pseudomizer** [comment]
I am sorry, but I opened this case and it is still not resolved for me and 2 other guys.

**Comment by bwssytems** [comment]
Sorry about that. You did get it to discover when you took your real hue bridge off the network. There is something the echo is doing in that case and I cannot fix that.

**Comment by bwssytems** [comment]
@simondsmason Please take a look at your windows firewall to make sure that upnp is not blocked.

**Comment by bwssytems** [comment]
@jlekhter Glad you fixed it with the port unblock.

**Comment by bwssytems** [comment]
So, this is not an issue list for Echo problems. This is an issue list for ha-bridge problems and enhancements and not really a forum. Hope you understand.

**Comment by Pseudomizer** [comment]
I don't know yet and I do understand.

**Comment by Pseudomizer** [comment]
I am happy to report that the issue has been resolved. It was the firmware of the Asus router. Replaced the stock firmware with merlin FW and that fixed the uPnP issues. 

The only devices working were Google Home and the HA bridge. Sonos, LIFX and Alexas had issues, which are now resolved with the new firmware. 

The hint to fix that came from LIFX support.  Very strange behavior that some worked and others didn't but glad it is fixed.

**Comment by snicker** [comment]
@Pseudomizer sorry to dig this one up, but what particular ASUS router were you using? I seem to be having the same problem and I am using the ASUS N66U, but it was already running the Merlin FW. I upgraded to 380.64 Merlin FW but the issues still persist. Did you have to change any specific settings? I have UPnP enabled and have all ports available internal and external.

**Comment by Pseudomizer** [comment]
RT5300 Asus. I installed the Merlin FW and left the default settings. Given the mount of devices on my home, I chose to reboot my house by cutting main power to avoid anything being cached by any device. After that, everything worked fine.

**Comment by mccum** [comment]
Hi all, sorry to get this thread on again, i am facing the same issue where my HA-BRIDGE is working, but alexa cant detect devices, my control bridge setting is on my PC ip address, which i am running 3.5.2 from and my asus-router have been reset and have uninstalled and reinstalled every app but it still does work. Any idea how to resolve?

**Comment by rafale77** [comment]
Most likely a router setting problem. Many of you have. Alexa uses UPnP. If your UPnP routing is blocked, not setup right in anyway, you will see issues. You probably want to go search for a UPnP help on your specific router.

**Comment by michaelhinchey** [comment]
Make sure the port that you are using is open as well.  I had an issue regarding this but it was a windows error.  I'm sure you have checked to make sure the port you are using is open.  Also make sure nothing else is competing with that port.  Ie. Kodi
Of course check the upnp settings first.

**Comment by mccum** [comment]
thanks @rafale77 @michaelhinchey . I did check the UPnP setting on my router and it was enabled. So is my port 80. The port is also not competing, since the bridge is also working, i am puzzled why Alexa is still not able to discover my Z-wave devices.

**Comment by bwssytems** [comment]
@mccum Did you turn on trace upnp to help determine if your bridge is getting any connection to your Alexa? That will help you decode what is happening.

**Comment by mccum** [comment]
@bwssytems yes, i got the warning and restarted everything, somehow it worked this time. i must have missed out something previously before i saved the settings. Thanks so much guys! its connected to my alexa and google home now! awesome!

**Comment by gregreid** [comment]
I just found what I think is a bug/limitation in the amazon-echo-ha-bridge emulator.  It seems that I can configure 28 devices maximum.  I'd originally configured about 45 of them and when I went to do the "Alexa, discover devices", I can see in my Putty terminal session (where I started the emulator) that Alexa was contacting the emulator properly, but after 30-ish seconds said "I wasn't able to discover any devices".  

It took a lot of experimenting to first go back to only 1 device configured -- which was discovered fine -- and then add a few more, with a new successful discovery, and so on until I got up to the "unable to discover any devices" problem again.  Then I deleted one at a time, with a fresh discovery each time, until Alexa was able to discover 28 devices.  I tried adding one more and none were discoverable.

I wonder if this is a known problem that might have a fix already?  I'm really tired now (long, long day at work) but will pick this up in the morning again.  And the first thing to do is obviously to make sure that I'm running the latest level of amazon-echo-ha-bridge code from GitHub.  

Greg

**Comment by audiofreak9** [comment]
This is a known limitation, and the limiting device is the Amazon Echo not the ha-bridge.

**Comment by gregreid** [comment]
Actually, I found and fixed the problem/limitation.  I was indeed using an older version of the amazon-echo-ha-bridge, version 0.2.1, which apparently had a 25 device limit.  (Somehow I managed to get 28 anyway.)

I upgraded to the latest 0.4.0 version, which allows 75 devices by default -- and more if needed, by a new command-line argument, in 25 device increments.  It actually wasn't quite as easy as that, because 0.4.0 has a BUG whereby it seemingly ignores any --server.port=xxxx you give it.  Therefore it will ONLY run using the default port of 8080 ... and also uses 8081 and 8082 with its default setup (25 devices on each sequential port).  

I had mine set to 8099 because my NAS (where I also have my bridge running) uses 8080 for its admin interface.  So I had to first reconfigure my NAS to use 9090 instead.  (That in turn broke my router port forwarding rules for remote NAS access ... and on and on.)

Anyway, the 0.4.0 version does work with its default port setup, and Alexa is now able to discover my 45-ish devices.  Yay!

**Comment by bwssytems** [comment]
@gregreid The bridge you are using is not this bridge. That one is maintained by armzilla. The bridge here that is now version 4.5.0 does not have any limitations and is quite different.

**Comment by gregreid** [comment]
Oh!, so sorry for my confusion, and thanks for straightening me out.  I originally found this topic by a simple Google search, and didn't pay attention to the "bwssystems/ha-bridge" header (obviously).  Indeed, I'm in the completely wrong GitHub area for the "armzilla/amazon-echo-ha-bridge" that I'm actually using.  Sorry for the interruptions.  :-/

**Comment by bwssytems** [comment]
You should try this bridge though. It started out from Armzilla's code base, but is much more now.

**Comment by orctrial** [comment]
Hi 

I know this is an old discussion but reading through it just want to understand what does the setting "upnp IP address " on the ha-bridge do? Does it broadcast itself on that IP?

**Comment by silopautomation** [comment]
hi,
Alexa is unable to discover my HA bridge. done all the settings correctly. kindly assist. here is the upnp log.
>>>>>


10-21-2017 13:34:11.180 | INFO | Traceupnp: sendUpnpNotify notifyTemplate is <<<NOTIFY * HTTP/1.1  HOST: 239.255.255.250:1900  CACHE-CONTROL: max-age=100  LOCATION: http://192.168.1.165:80/description.xml  SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0  NTS: ssdp:alive  hue-bridgeid: B827EBFFFEBB53EB  NT: uuid:2f402f80-da50-11e1-9b23-b827ebbb53eb  USN: uuid:2f402f80-da50-11e1-9b23-b827ebbb53eb    >>> | com.bwssystems.HABridge.upnp.UpnpListener
-- | -- | -- | --
10-21-2017 13:34:11.182 | INFO | Traceupnp: sendUpnpNotify notifyTemplate is <<<NOTIFY * HTTP/1.1  HOST: 239.255.255.250:1900  CACHE-CONTROL: max-age=100  LOCATION: http://192.168.1.165:80/description.xml  SERVER: Linux/3.14.0 UPnP/1.0 IpBridge/1.15.0  NTS: ssdp:alive  hue-bridgeid: B827EBFFFEBB53EB  NT: uuid:2f402f80-da50-11e1-9b23-b827ebbb53eb  USN: uuid:2f402f80-da50-11e1-9b23-b827ebbb53eb    >>> | com.bwssystems.HABridge.upnp.UpnpListener

**Comment by silopautomation** [comment]
on above...my bridge ip is 192.168.165
upnp enabled on router
wont know why alexa is not discovering the devices

**Comment by bwssytems** [comment]
What type of Echo device do you have?

**Comment by silopautomation** [comment]
Using Echo top version,..see the image..
 
Amazon Echo - Black (1st Generation)

**Comment by silopautomation** [comment]
Top model of Echo....

**Comment by klccjeremy** [comment]
Hi,

I noticed that router settings seems to be a possible cause. I'm not able to discover devices in my home, but when I bring the whole thing to my fren's house, it miraculously discover all devices. 

I've search in google and seems that UPnP settings might affect it. I turned on UPnP on my router, but it still doesnt work. Any advice?

This is my log:

[4.5.6.txt](https://github.com/bwssytems/ha-bridge/files/1522488/4.5.6.txt)

**Comment by michaelhinchey** [comment]
Have u tried changing the upnp address to your ip address?
Instead of the prefilled








    Every dream is a goal with a date on it; if you can conceive it believe
it, you can achieve it. Don't let anyone steal your dream. Make the
decision to take a positive attitude, take pride in yourself and what you
do, work hard and stay with it. A mind is a terrible thing to waste.


"I felt bad because I couldn't wheelie; until I met a man with no bicycle"

"You are young at any age... if you are planning on tomorrow"

**Comment by klccjeremy** [comment]
@michaelhinchey 
yeah.. i filled up the raspberry pi's address as the upnp address.

**Comment by aradia-1989** [comment]
Well damn.. I also have the ASUS ac5300 running on stock FW... pseudo or anyone else, do you know if it’s fixed or do I absolutely have to get Merlin on here? Alexa can discover smart plugs and nests and such, but doesn’t discover my harmony hub or Rachio for example..
